figures/pipeline/course_daily_metrics.py

```diff
@@ -145,24 +146,25 @@ def get_days_to_complete(course_id, date_for):
     When we have to support scale, we can look into optimization
     techinques.
     """
-    certificates = GeneratedCertificate.objects.filter(
+    grades = PersistentCourseGrade.objects.filter(
         course_id=as_course_key(course_id),
-        created_date__lte=as_datetime(date_for))
+        passed_timestamp__isnull=False
+    )
 
     days = []
     errors = []
-    for cert in certificates:
-        ce = CourseEnrollment.objects.filter(
+    for grade in grades:
+        course_enrollment = CourseEnrollment.objects.filter(
```

Comment:
we can simplify first course enrollment logic like this
```
course_enrollment = CourseEnrollment.objects.filter(
            course_id=as_course_key(course_id),
            user__id=grade.user_id)
).first()
```

Reply:
I was following appsembler's pattern but I think I can simplify/improve it while adding the errors as well. let me do it.

Reply:
improved it

Reply:
Apparently appsembler wants to track errors if multiple enrollments are found against a student in a course. I’m trying to conserve that pattern. If u think it’s better to remove that part altogether then I can do that